### DISCUSSION TITLE
Bash Suggestions for gsd-tools/local-scripts/update-suse-urls

### DISCUSSION THREAD

**Discussion by jasnow** [cloudsecurityalliance/gsd-tools#182] [Show and tell]
FYI: Starting up "sed" once is faster than multiple times.

```sh
time echo "001" | sed -e "s/001/002/" -e "s/002/003/" -e "s/003/004/" 

time echo "001" | sed -e "s/001/002/" |sed --e "s/002/003/" |sed -e "s/003/004/" 
```

Note the single "sed" execute the "-e" left-to-right.

---------------------------------------------------------------

Also (defensive programming)

```sh
 if [ "X$FOUND" == "Xtrue" ]
```

handles when $FOUND is empty better than

```sh
if [ $FOUND == "true" ]
```

---------------------------------------------------------------

Test example:

```sh
FOUND=""

### "X"
if [ "X$FOUND" = "Xtrue" ] ; then
    echo "Got true"
else
    echo "Got not true"
fi

#### No "X"
if [ $FOUND = "true" ] ; then
    echo "Got true"
else
    echo "Got not true"
fi
```

**Comment by joshbuker** [comment]
@jasnow Do you want to take a stab at updating this via PR? Contributions welcome! :smile: 

I'm trying to get our contribution guide / getting started easier to follow, so let me know if you run into stumbling blocks.

**Reply by jasnow** [reply]
Here's the 2 changes - you can test them before and after - https://gist.github.com/jasnow/fc1a3cf78fb709bcd4e81fb4d575741a
